teep.cddl

```diff
@@ -0,0 +1,144 @@
+teep-message = $teep-message-type .within teep-message-framework
+
+SUIT-envelope = bytes ; placeholder
+
+teep-message-framework = [
+  type: 0..23 / $teep-type-extension,
+  token: uint,
+  options: { * teep-option },
+  * int; further integers, e.g. for data-item-requested
+]
+
+teep-option = (uint => any)
+
+; this is cleaned up so:
+; -- type is always 0..23, holler if that is not enough
+; -- $teep-type-extension is a possibility to extend that later, is that desirable?
+; -- data-items come after the option, so that has a fixed position
+;    and the options info is available when decoding data items.  But then
+;    I don't understand what the point of data-items is.
+; -- we did not use generics here, even though that might reduce some of the clutter.
+
+; messages defined below:
+$teep-message-type /= query-request
+$teep-message-type /= query-response
+$teep-message-type /= trusted-app-install
+$teep-message-type /= trusted-app-delete
+$teep-message-type /= teep-error
+$teep-message-type /= teep-success
+
+; message type numbers
+TEEP-TYPE-query-request = 1
+TEEP-TYPE-query-response = 2
+TEEP-TYPE-trusted-app-install = 3
+TEEP-TYPE-trusted-app-delete = 4
+TEEP-TYPE-teep-error = 5
+TEEP-TYPE-teep-success = 6
+
+version = uint .size 4
+ext-info = uint 
+
+; data items as bitmaps
+data-item-requested = $data-item-requested .within uint .size 8
+attestation = 1
+$data-item-requested /=  attestation
+trusted-apps = 2
+$data-item-requested /= trusted-apps
+extensions = 4
+$data-item-requested /= extensions
+suit-commands = 8
+$data-item-requested /= suit-commands
+
+query-request = [
+  type: TEEP-TYPE-query-request,
+  token: uint,
+  options: {
+    ? supported-cipher-suites => suite,
+    ? nonce => bytes,
```

Comment:
Using the same size as EAT.
? nonce => bytes .size (8..64),

Reply:
Good catch. Thanks. I knew that, but forgot it :-)